gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceSource.java

```diff
@@ -209,7 +212,27 @@ private Histogram getHistogram(String entity, String watermarkColumn, SourceStat
     }
 
     for (int year = startYear; year <= endYear; year++) {
-      values.put("year", Integer.toString(year));
+
+      if (year == startYear) {
+        values.put("start", lowWatermarkDate);
+        values.put("greater", partition.isLowWatermarkInclusive() ? ">=" : ">");
+      } else {
+        calendar.clear();
```

Comment:
We can define a help function which returns the time string given a year integer

Reply:
Created getDateString helper